ipaserver/install/server/replicainstall.py

```diff
@@ -1252,11 +1253,9 @@ def promote_check(installer):
             config.kra_host_name = kra_host
             kra_enabled = True
         else:
-            if options.setup_kra:
-                root_logger.error("There is no KRA server in the domain, "
-                                  "can't setup a KRA clone")
-                raise ScriptError(rval=3)
             kra_enabled = False
+            if options.setup_kra:
+                kra_enabled = True
 
```

Comment:
wee comment about options.setup_kra for the other file, we can completely remove the kra_enabled variable everywhere and these 2 lines and simply set installer.setup_kra = options.setup_kra at the end of the check phase (note you have to do this for both classic install and promotion, and replace checks of kra_enabled with options.setup_kra throughout this file)